src/cli/rustup_mode.rs

```diff
@@ -1281,8 +1281,25 @@ async fn toolchain_link(
 }
 
 fn toolchain_remove(cfg: &mut Cfg<'_>, opts: UninstallOpts) -> Result<utils::ExitCode> {
+    let default_toolchain = cfg.get_default().ok().flatten();
+    let active_toolchain = cfg.find_active_toolchain().ok().flatten().map(|(it, _)| it);
+
     for toolchain_name in &opts.toolchain {
         let toolchain_name = toolchain_name.resolve(&cfg.get_default_host_triple()?)?;
+
+        if active_toolchain
+            .as_ref()
+            .is_some_and(|n| n == &toolchain_name)
+        {
+            warn!("it seems that you are removing the active toolchain, is this intentional?");
```

Comment:
For a non-prompt warning, let's get rid of the question mark? Also is there a reason to soften the language with "it seems that"? I think this could just be "removing the active toolchain"?

Reply:
@djc I want to attract the user's attention and at least indicate that they are doing something probably undesirable. I've adjusted the wording nonetheless.